maps/biter_battles_v2/init.lua

```diff
@@ -179,6 +183,27 @@ function Public.draw_structures()
 	--Terrain.generate_spawn_goodies(surface)
 end
 
+local reveal_token = Token.register(
+    function()
+		if global.bb_settings["bb_map_reveal_toggle"] then
+			local surface = game.surfaces[global.bb_surface_name]
+			local radius = 2000
+			game.forces["north"].chart(surface, {{0-radius, 500}, {0+radius, -500}})
+			game.forces["south"].chart(surface, {{0-radius, 500}, {0+radius, -500}})
+		end
+    end
+)
+
+function Public.reveal_map()
+	if global.bb_settings["bb_map_reveal_toggle"] then
+		local surface = game.surfaces[global.bb_surface_name]
+		local radius = 2000
+		game.forces["north"].chart(surface, {{0-radius, 500}, {0+radius, -500}}) 
+		game.forces["south"].chart(surface, {{0-radius, 500}, {0+radius, -500}}) 
+		Task.set_timeout_in_ticks(10800, reveal_token)
```

Comment:
```suggestion
		local width = 2000 -- for one side
		local height = 500 -- for one side
		for x = 16, width, 32 do
			for y = 16, height, 32 do
				game.forces["spectator"].chart(surface, {{-x, -y}, {-x, -y}})
				game.forces["spectator"].chart(surface, {{x, -y}, {x, -y}})
				game.forces["spectator"].chart(surface, {{-x, y}, {-x, y}})
				game.forces["spectator"].chart(surface, {{x, y}, {x, y}})
			end
		end
```
in this way rechart is not required

Reply:
Wow, what a great update !  Thanks a lot @developer-8  I was wondering if we could have avoid this double rechart... You found it.